### PR TITLE
Support valid_time and the like for queries in xtdb tools

### PR DESCRIPTION
**Pull request by originalsouth**
### Changes
Support valid_time and the like for queries in xtdb tools

### Issue link
N/A

### Demo
Make a setup with some objects.

Verify there are some objects:
`octopoes/tools/xtdb-cli.py query '{:query {:find [(pull ?var [*])] :where [[?var :object_type]]}}' | jq -C | less -R`

Verify there weren't any objects 91 years ago (you can place other dates here):
`octopoes/tools/xtdb-cli.py query --valid-time 1933-2-27 '{:query {:find [(pull ?var [*])] :where [[?var :object_type]]}}' | jq -C | less -R`

### QA notes
Run the demo; perhaps change valid-time value to something around the creation of the objects so you can see the history.

---

### Code Checklist

- [x] All the commits in this PR are properly PGP-signed and verified.
- [x] This PR only contains functionality relevant to the issue.
- [ ] I have written unit tests for the changes or fixes I made.
- [x] I have checked the documentation and made changes where necessary.
- [x] I have performed a self-review of my code and refactored it to the best of my abilities.

- [ ] Tickets have been created for newly discovered issues.
- [x] For any non-trivial functionality, I have added integration and/or end-to-end tests.
- [x] I have informed others of any required `.env` changes files if required and changed the `.env-dist` accordingly.
- [x] I have included comments in the code to elaborate on what is not self-evident from the code itself, including references to issues and discussions online, or implicit behavior of an interface.

---

## Checklist for code reviewers:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_code.md) into your comment.

---

## Checklist for QA:

Copy-paste the checklist from [the docs/source/templates folder](https://github.com/minvws/nl-kat-coordination/blob/main/docs/source/templates/pull_request_template_review_qa.md) into your comment.

### CODE REVIEW

**Review by ammar92 — APPROVED** [review]
Looks good to me

**Comment by stephanie0x00** [comment]
## Checklist for QA:

- [x] I have checked out this branch, and successfully ran a fresh `make reset`.
- [x] I confirmed that there are no unintended functional regressions in this branch:
  - [x] I have managed to pass the onboarding flow
  - [x] Objects and Findings are created properly
  - [x] Tasks are created and completed properly
- [x] I confirmed that the PR's advertised `feature` or `hotfix` works as intended.
- [x] I checked the logs for errors and/or warnings and made issues where necessary

### What works:
Looks good. The commands work as expected. 

### What doesn't work:
n/a

### Bug or feature?:
n/a

File: octopoes/tools/xtdb-cli.py [hunk 1]
```diff
@@ -61,15 +61,24 @@ def status(ctx: click.Context):
 
 
 @cli.command(help='EDN Query (default: "{:query {:find [ ?var ] :where [[?var :xt/id ]]}}")')
+@click.option("--tx-id", type=int, help="Defaulting to latest transaction id (integer)")
+@click.option("--tx-time", type=click.DateTime(), help="Defaulting to latest transaction time (date)")
+@click.option("--valid-time", type=click.DateTime(), help="Defaulting to now (date)")
```
[the remark this answers is not in the document]
Comment (originalsouth) [reply octopoes/tools/xtdb-cli.py:66]: We are assuming all datetime revolving around Octopoes are UTC -- I think we take care of that? Would you like to add that to the documentation?

Comment (underdarknl) [inline octopoes/tools/xtdb-cli.py:66]:
```suggestion
@click.option("--tx-time", type=click.DateTime(), help="In UTC, Defaulting to latest transaction time (date)")
@click.option("--valid-time", type=click.DateTime(), help="In UTC, Defaulting to now (date)")
```

**Review by underdarknl — APPROVED** [review]

**Pull request merged**